### PR TITLE
[llvm][lld] Support R_RISCV_GOT32_PCREL

### PR DESCRIPTION
**Pull request by PiJoules**
This is the followup implementation to
https://github.com/riscv-non-isa/riscv-elf-psabi-doc/pull/402 that supports this relocation in llvm and lld.

### CODE REVIEW

**Comment by PiJoules** [comment]
Didn't notice until now that 41 was taken up by `R_RISCV_GNU_VTINHERIT` which I believe is a very old relocation that's unsupported by most linkers along with `R_RISCV_GNU_VTENTRY`. Not sure if we want to remove them altogether or change their codes or give `R_RISCV_GOT32_PCREL` a new code. For now, I just removed them.

File: lld/test/ELF/riscv64-reloc-got32_pcrel.s [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// REQUIRES: riscv
+// RUN: llvm-mc -filetype=obj -triple=riscv64 %s -o %t.o
+// RUN: ld.lld %t.o -o %t.so -shared
+// RUN: llvm-readobj -S %t.so | FileCheck --check-prefix=SEC %s
+// RUN: llvm-objdump --no-print-imm-hex -s -d %t.so | FileCheck %s
```
[the remark this answers is not in the document]
Comment (PiJoules) [reply lld/test/ELF/riscv64-reloc-got32_pcrel.s:5]: Added overflow test and using llvm-readelf.

File: lld/test/ELF/riscv64-reloc-got32_pcrel-overflow.s [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// REQUIRES: riscv
```
Comment (MaskRay) [inline lld/test/ELF/riscv64-reloc-got32_pcrel-overflow.s:1]: Similar to the aarch64 patch, this negative test should be merged into `lld/test/ELF/riscv64-reloc-got32_pcrel.s`

I think we use `_` or `-` consistently. If there is a `-`, just use `-` for `_` in `got32_pcrel` as well

**Comment by asb** [comment]
@PiJoules happy new year! Might you have a moment to address @Maskray's review feedback?

**Comment by PiJoules** [comment]
> @PiJoules happy new year! Might you have a moment to address @MaskRay's review feedback?

Done. Sorry, just got back from vacation then got sick for a bit :/

**Review by MaskRay — APPROVED** [review]

**Pull request merged**